src/target.zig

```diff
@@ -398,6 +398,31 @@ pub fn addrSpaceCastIsValid(
     }
 }
 
+/// Under SPIR-V with Vulkan, pointers are not 'real' (physical), but rather 'logical'. Effectively,
+/// this means that all such pointers have to be resolvable to a location at compile time, and places
+/// a number of restrictions on usage of such pointers. For example, a logical pointer may not be
+/// part of a merge (result of a branch) and may not be stored in memory at all. This function returns
+/// for a particular architecture and address space wether such pointers are logical.
+pub fn arePointersLogical(target: std.Target, as: AddressSpace) bool {
+    if (target.os.tag != .vulkan) {
+        return false;
+    }
```

Comment:
I get that this is obviously the case for Vulkan specifically, but I wonder if checking the OS is really the right approach; we have `spirv` and `spirv32`/`spirv64` arch tags, and it's my understanding that `spirv` (that is, no bitness) implies logical pointers. Is that correct? If so, that seems like the 'more correct' thing to check for.

Reply:
The spirv64-vulkan arch currently maps to logical pointers and 64-bit physical storage buffer pointers. In theory those pointers can still be shoved under the logical target, but I'm not convinced of is usefulness in general